control/matlab/wrappers.py

```diff
@@ -53,19 +56,71 @@ def bode(*args, **keywords):
         * >>> bode(sys1, sys2, ..., sysN, w)
         * >>> bode(sys1, 'plotstyle1', ..., sysN, 'plotstyleN')
     """
+    from ..freqplot import bode_plot
 
-    # If the first argument is a list, then assume python-control calling format
-    from ..freqplot import bode as bode_orig
+    # If first argument is a list, assume python-control calling format
     if (getattr(args[0], '__iter__', False)):
-        return bode_orig(*args, **keywords)
+        return bode_plot(*args, **kwargs)
 
-    # Otherwise, run through the arguments and collect up arguments
-    syslist = []; plotstyle=[]; omega=None;
+    # Parse input arguments
+    syslist, omega, args, other = _parse_freqplot_args(*args)
+    kwargs.update(other)
+
+    # Call the bode command
+    return bode_plot(syslist, omega, *args, **kwargs)
+
+
+def nyquist(*args, **kwargs):
+    """nyquist(syslist[, omega])
+
+    Nyquist plot of the frequency response
+
+    Plots a Nyquist plot for the system over a (optional) frequency range.
+
+    Parameters
+    ----------
+    sys1, ..., sysn : list of LTI
+        List of linear input/output systems (single system is OK).
+    omega : array_like
+        Set of frequencies to be evaluated, in rad/sec.
+
+    Returns
+    -------
+    real : ndarray (or list of ndarray if len(syslist) > 1))
+        real part of the frequency response array
+    imag : ndarray (or list of ndarray if len(syslist) > 1))
+        imaginary part of the frequency response array
+    omega : ndarray (or list of ndarray if len(syslist) > 1))
+        frequencies in rad/s
+
+    """
+    from ..freqplot import nyquist_plot
+
+    # If first argument is a list, assume python-control calling format
+    if (getattr(args[0], '__iter__', False)):
```

Comment:
I have seen this style elsewhere in the library so I am curious: isn’t hasattr(args[0], ‘__iter__’) both more readable and compact?

Reply:
`hasattr` also returns True if `args[0]` is an iterable sequence, but empty. This code does only branch if `args[0]` is not iterable or the sequence does not cast to False (is empty).

Reply:
I see, thanks!

Reply:
`getattr([], '__iter__', False)` returns ` <method-wrapper '__iter__' of list object ...`.  So looks like both of them would follow the branch?